lms/security.py

```diff
@@ -143,19 +143,29 @@ def _get_lti_user(request):
         partial(bearer_token_schema.lti_user, location="form"),
         OAuthCallbackSchema(request).lti_user,
     ]
-    # Avoid checking for the LTI1.3 JWT if not there
-    if "id_token" in request.params:
-        # Don't replace all auth methods in case somehow we have a bogus token
-        # but try this one first.
-        schemas.insert(0, LTI13AuthSchema(request).lti_user)
 
     lti_user = None
-    for schema in schemas:
+    # Avoid checking for the LTI1.3 JWT if not there
+    if "id_token" in request.params:
```

Comment:
In the previous version if for some reason a valid request for any other method had, for some reason, also an "id_token" we'd first validate that and then continue with the rest.

Here we just look at the id_token if present. 

I don't think we ever expected the old behavior to happen.